src/engine/UNSATCertificate.h

```diff
@@ -0,0 +1,282 @@
+/*********************                                                        */
+/*! \file UNSATCertificate.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Omri Isac, Guy Katz
+ ** This file is part of the Marabou project.
+ ** Copyright (c) 2017-2022 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved. See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** [[ Add lengthier description here ]]
+ **/
+
+
+#ifndef __UNSATCertificate_h__
+#define __UNSATCertificate_h__
+
+#include "BoundExplainer.h"
+#include "SmtLibWriter.h"
+#include "PiecewiseLinearFunctionType.h"
+#include "ReluConstraint.h"
+
+/*
+  Contains all necessary info of a ground bound update during a run (i.e from ReLU phase-fixing)
+*/
+struct PLCExplanation
+{
+    unsigned _causingVar;
+    unsigned _affectedVar;
+    double _bound;
+    bool _isCausingBoundUpper;
+    bool _isAffectedBoundUpper;
+    double *_explanation;
+    PiecewiseLinearFunctionType _constraintType;
+    unsigned _decisionLevel;
+
+    ~PLCExplanation()
+    {
+        if ( _explanation )
```

Comment:
Memory leak: add a constructor where _explanation is initialized to null. Also, move the destructor implementation to cpp.